docs/tutorial/parameter-types/pydantic-types.md

```diff
@@ -0,0 +1,84 @@
+Pydantic types such as [AnyUrl](https://docs.pydantic.dev/latest/api/networks/#pydantic.networks.AnyUrl) or [EmailStr](https://docs.pydantic.dev/latest/api/networks/#pydantic.networks.EmailStr) can be very convenient to describe and validate some parameters.
```

Comment:
Perhaps we should not focus the documentation and unit tests so much on `EmailStr`, as this requires yet another package, `email-validator`. I noticed this is taken into account by adding `pip install pydantic[email]` to `pyproject.toml`, but a few lines down in this file, it says to alternatively `pip install pydantic` which won't pull in `email-validator`. 

In general, we need to make sure that the unit tests and other functionality exits gracefully and not crashes with a long stack trace and/or failing unit tests if not all optional libraries are installed.

Reply:
Well, for me email is one of the primary use cases for which I took interest into making this pull request.
I can agree to using urls for the tests but I would like to leave it in docs if you agree

Reply:
I think that'd be a good solution: use URLs in the example code & unit tests, and not install `email-validator` by default. Then mention it in the documentation and add a note that it requires `email-validator` as well (or not - as this note is already prominent on the Pydantic page you're linking)